### PR TITLE
Counter cache hit miss metrics

### PR DESCRIPTION
**Pull request by danielnorberg**

### CODE REVIEW

**Comment by danielnorberg** [comment]
@honnix PTAL

File: styx-api-service/src/main/java/com/spotify/styx/api/ResourceResource.java [hunk 1]
```diff
@@ -46,11 +45,9 @@
   static final String BASE = "/resources";
 
   private final Storage storage;
-  private ShardedCounter shardedCounter;
 
-  public ResourceResource(Storage storage, ShardedCounter shardedCounter) {
+  public ResourceResource(Storage storage) {
     this.storage = Objects.requireNonNull(storage);
-    this.shardedCounter = shardedCounter;
```
Comment (honnix) [inline styx-api-service/src/main/java/com/spotify/styx/api/ResourceResource.java]: waat? interesting.

File: styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 2]
```diff
@@ -160,6 +166,8 @@
   private final ConcurrentMap<String, Meter> workflowConsumerMeters;
   private final ConcurrentMap<String, Histogram> tickHistograms;
   private final ConcurrentMap<Tuple2<String, String>, Meter> datastoreOperationMeters;
+  private final Meter counterCacheHitMeter;
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: It seems we have convention to put simple stuff together before those maps.

File: styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 3]
```diff
@@ -192,6 +200,8 @@ public MetricsStats(SemanticMetricRegistry registry, Time time) {
     this.workflowConsumerMeters = new ConcurrentHashMap<>();
     this.tickHistograms = new ConcurrentHashMap<>();
     this.datastoreOperationMeters = new ConcurrentHashMap<>();
+    this.counterCacheHitMeter = registry.meter(COUNTER_CACHE_RATE.tagged(COUNTER_CACHE_RESULT, COUNTER_CACHE_HIT));
```
Comment (honnix) [inline styx-common/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: It seems we have convention to put simple stuff together before those maps.

**Comment by honnix** [comment]
a few nitpickings, otherwise LGTM.

**Comment by honnix** [comment]
:shipit:

**Pull request merged**